### ISSUE TITLE
Feature Request: Skip provisioning after generating keys

### ISSUE THREAD

**Issue by codepuncher** [roots/trellis-cli#291]
### Terms

- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This request is not a duplicate of an existing issue
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Summary

Skip environment provisioning with a flag e.g. `--no-provision`.
Useful when using Trellis only for deployment (e.g. on managed hosts like Kinsta).

#### Why are we doing this?
To allow skipping environment provisioning after a key has been generated.

#### What use cases does it support?
For those using Trellis with managed hosts, there is no need for provisioning since the host is not configured by Trellis.
Or perhaps there is another step in your workflow that needs to happen before provisioning, so skipping this would be beneficial.

#### What is the expected outcome?
To skip provisioning the host.

#### Potential conflicts / foreseeable issues
N/A
